### ISSUE TITLE
Boolean issue

### ISSUE THREAD

**Issue by luc-boussant** [ngrunwald/datasplash#101]
Hi ! I am currently having some issues with some of my pipelines regarding the boolean fields. 
For example, i have : 
```
(let [p (ds/make-pipeline {})]
    (ds/->> :log-values p
            (ds/generate-input [{:deleted false} {:deleted true}])
            (ds/map #(log/error %) {:name "do-log"}))
    (ds/run-pipeline p))
;; => {:deleted true}
;; => {:deleted true}
```
Does anyone have the same problem and/or has anyone found some way to bypass the issue ?

Thanks ;)

**Comment by xfthhxk** [comment]
I can reproduce the same thing.  If I change the values to integers or strings it works as expected.

```clj
(->> (ds/make-pipeline {})
         (ds/generate-input [{:deleted "false"} {:deleted "true"}])
         (ds/map #(log/info %) {:name "do-log"})
         (ds/run-pipeline)
         (ds/wait-pipeline-result))

;; => {:deleted "true"}  
;; => {:deleted "false"}
```

**Comment by RolT** [comment]
This is an issue with nippy / java serialization

```clojure
(-> (Boolean. false) nippy/freeze nippy/thaw)

;; => true
```

I tried overriding the serialization and it seems to work:

```clojure
(nippy/extend-freeze Boolean :my/bool [x data-output] (.writeUTF data-output (if (boolean x) "true" "false")))
(nippy/extend-thaw :my/bool [data-input] (if (= "true" (.readUTF data-input)) true false))
```

**Comment by edporras** [comment]
@RolT thanks so much. I saw your [nippy PR](https://github.com/ptaoussanis/nippy/pull/150). I'll wait a few days to see if Peter pushes a new nippy release.

**Comment by edporras** [comment]
Fix in nippy 3.1.3
